### DISCUSSION TITLE
How to get the Zwave Driver status

### DISCUSSION THREAD

**Discussion by sebekhtc** [zwave-js/zwave-js-ui#3647] [Q&A]
Once a month my zwave stick (Gen 5+) is not responding showing similar (taken from another thread):
![305048567-ec6eceeb-f1ea-46dc-9c0e-37ce71d66760](https://github.com/zwave-js/zwave-js-ui/assets/5500978/dcd758fb-f2ae-43d9-843d-de75574f8506)

In such case the ls shows that device is connected and docker restart fixes it.
I would like to restart it whenever my cron python script detects Driver NOT ready.

How to pull the shown above driver info using eg. curl etc in linux (Raspberry Pi)?

**Comment by robertsLando** [comment]
you could use the health api: https://zwave-js.github.io/zwave-js-ui/#/guide/healthcheck?id=health-check-endpoints

**Reply by sebekhtc** [reply]
thanks a lot!
I remember that 
curl localhost:8091/health/zwave -H "Accept: text/plain"
showed me OK as a response in driver error/not ready state.
Will try the wget next time and check numbers received.

**Reply by robertsLando** [reply]
Let me know

**Comment by robertsLando** [comment]
@sebekhtc I have just submitted a possible fix that could have caused the curl command to not return the correct status. Please if you are using docker switch to `master` tag once the build finish: https://github.com/zwave-js/zwave-js-ui/actions/runs/8449004280/job/23142192398

**Reply by sebekhtc** [reply]
will try what suggested above.
I tried now to just unplug the dongle and got this:
_pi@raspberrypi:/tmp $ curl localhost:8091/health/zwave -H "Accept: text/plain"
Errorpi@raspberrypi:/twget --no-verbose --spider --no-check-certificate --header "Accept: text/plain" http://192.168.1.6:8091/health
http://192.168.1.6:8091/health:
http://192.168.1.6:8091/health:
Remote file does not exist -- broken link!!!_

but the difference I noticed is the notification bell when hover with mouse shows Controller is Idle but for the past cases it was just a question mark.
Have no idea how to recreate that state when controller can't be detected but exists in the system if I issue ls command to list it.

**Reply by robertsLando** [reply]
> Have no idea how to recreate that state when controller can't be detected but exists in the system if I issue ls command to list it.

I think it may be something related to soft reset and that happens only on startup, another solution could be to disable soft reset on startup (from settings) even if that's not ideal

**Reply by sebekhtc** [reply]
ok, I have it enabled, will disable and wait if issue reoccurres.

**Reply by sebekhtc** [reply]
BTW would it be possible to add for example lastseen epoch time to controller node in a node json file (or something like not active)?
For some reason nodes have the last seen (example node.176.lastSeen) but not the controller.

**Reply by robertsLando** [reply]
@sebekhtc that makes no sense because controller is up or down there is no real "last seen"

**Reply by sebekhtc** [reply]
had no idea about that... thought is always active. Anyway I can see that master has been built. going to update.

**Reply by robertsLando** [reply]
Just to explain better: Controller is your zwave antenna, each information you see on UI passes through it, that's why having a last seen on it just have no sense :)

**Reply by sebekhtc** [reply]
I'm wondering why other nodes have last seen while controller does not. Is it the same as Last Active column at the zwave control panel? All nodes including controller have it.

if i sort the nodes by the Last Active, the controller is active (as TX/RX is ongoing) like the other nodes, and controller stays on the top (second or third place as responding to it the devices come to the top). 
If I can see that controller's last activity value is hour(s) from current time there should be something wrong, shouldn't be?

I see that controller debug info has a value lastActive:
_"lastActive": 1711555323039,_
that is live, how to read it by curl/wget etc? Where is that file located?

**Reply by robertsLando** [reply]
it is there but as said you should ignore it. Use the helath endpoints them are reliable

**Reply by sebekhtc** [reply]
ok, I will monitor the situation together with my script checking the last active time for all connected devices based on the command I developed to get the highest last active Epoch time
`cat /opt/zwavejsui/store/ce758003.jsonl | jq -c 'select(.k | endswith("lastSeen")) |.v' | sort -un | tail -1  `
